src/components/SearchMap/SearchMapWithGoogleMaps.js

```diff
@@ -83,11 +91,28 @@ export const isMapsLibLoaded = () =>
   typeof window !== 'undefined' && window.google && window.google.maps;
 
 /**
- * FIX "TypeError: Cannot read property 'overlayMouseTarget' of null"
- * Override draw function to catch errors with map panes being undefined to prevent console errors
- * https://github.com/tomchentw/react-google-maps/issues/482
+ * To render HTML on top of Google Maps, we need use OverlayView.
+ * Note: we don't extend the OverlayView directly, because then
+ * the class should be defined inside initializeMap function.
  */
-class CustomOverlayView extends OverlayView {
+class CustomOverlayView extends Component {
+  constructor(props, context) {
+    super(props, context);
+    const overlayView = new window.google.maps.OverlayView();
+
+    overlayView.onAdd = this.onAdd.bind(this);
+    overlayView.draw = this.draw.bind(this);
+    overlayView.onRemove = this.onRemove.bind(this);
+    this.onPositionElement = this.onPositionElement.bind(this);
+
+    // You must call setMap() with a valid Map object to trigger the call to
+    // the onAdd() method and setMap(null) in order to trigger the onRemove() method.
+    overlayView.setMap(props.map);
+    this.state = {
+      [OVERLAY_VIEW]: overlayView,
```

Comment:
Why does the key have to be dynamic here? If anyone wants to add new state to this component, they have to look carefully what the actual value of `OVERLAY_VIEW` is to know what key is used in the state object.

Reply:
Good point! It's some old code from react-google-maps keeping an old bugfix closer to library's codebase for potential upstream updates. Removed.